node/src/actors/session/handlers.rs

```diff
@@ -794,31 +821,50 @@ fn check_beacon_compatibility(
         });
     }
 
-    match current_beacon
-        .highest_block_checkpoint
+    let my_beacon = if let Some(target_beacon) = target_superblock_beacon {
+        target_beacon
+    } else {
+        current_beacon.highest_superblock_checkpoint
+    };
```

Comment:
Not a big blocker, but I'd appreciate a comment on what's the rationale behind this new argument and this statement.

Reply:
@lrubiorod you can summarize this into a comment:

Set the target superblock beacon.

When the superblock consensus is different from the superblock created by this node, the process is to revert and ask the outbound peers for new blocks. We use the target beacon to ensure that the outbound peers have the correct superblock.

Note that this target is only used if the received_beacon checkpoint is equal to the target superblock checkpoint: this means that this check is only valid during one superepoch, if the node fails to find enough peers with the target superblock beacon, it will fall back to the old behavior and allow connections to any peers, ignoring the target superblock beacon.

Reply:
Not really, imagine that the SuperBlock 24 is achieved with the hash "aaaa..." by more than 2/3 of the signing committee, but you are in a fork so... you will revert and your last_superblock consolidated is 23 but your superblock_target is 24. If during the next handshakings the other peers send a LastBeacon with the superblock index 23 or index 24 but with different hash to "aaaa...", they will be rejected.

It only works with the old behaviour when it doesn't have a target, and it happens in "Same as local", "Not Reached" and "Unknown"

Reply:
Right